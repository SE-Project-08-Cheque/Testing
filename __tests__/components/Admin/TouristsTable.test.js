// import dependencies
import React from "react";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import react-testing methods
import { render, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import the component for testing
import TouristsTable from "../../../components/Admin/TouristsTable";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <TouristsTable
      tourists={[
        {
          firstname: "chandima",
          lastname: "Amarasena",
        },

        {
          firstname: "Hirumal",
          lastname: "Perera",
        },
      ]}
    />,
    div
  );
});

it("renders tourists table correctly", () => {
  const { getAllByTestId } = render(
    <TouristsTable
      tourists={[
        {
          firstname: "chandima",
          lastname: "amarasena",
        },

        {
          firstname: "Hirumal",
          lastname: "Perera",
        },
      ]}
    />
  );

  const names = getAllByTestId("name");
  expect(names[0]).toHaveTextContent("chandima amarasena");
  expect(names[1]).toHaveTextContent("Hirumal Perera");
 });

it("matches snapshot", () => {
  const tree = renderer.create(
    <TouristsTable
      tourists={[
        {
          firstname: "chandima",
          lastname: "amarasena",
        },

        {
          firstname: "Hirumal",
          lastname: "Perera",
        },
      ]}
    />
  );

  expect(tree).toMatchSnapshot();
});
