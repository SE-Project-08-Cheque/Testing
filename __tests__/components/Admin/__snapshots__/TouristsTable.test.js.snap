// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-vrubjm"
>
  <table
    className="chakra-table css-0"
    role="table"
  >
    <thead
      className="css-0"
    >
      <tr
        className="css-0"
        role="row"
      >
        <th
          className="css-0"
        >
          Profile photo
        </th>
        <th
          className="css-0"
        >
          Name
        </th>
        <th
          className="css-0"
        >
          Reviews
        </th>
        <th
          className="css-0"
        >
          State
        </th>
        <th
          className="css-0"
        >
          Restrict the user
        </th>
      </tr>
    </thead>
    <tbody
      className="css-0"
    >
      <tr
        className="css-0"
        role="row"
      >
        <td
          className="css-0"
          role="gridcell"
        >
          <span
            className="chakra-avatar css-y4jt82"
          >
            <div
              aria-label="user"
              className="chakra-avatar__initials css-0"
            >
              u
            </div>
          </span>
        </td>
        <td
          className="css-0"
          data-testid="name"
          role="gridcell"
        >
          chandima
           
          amarasena
           
        </td>
        <td
          className="css-0"
          role="gridcell"
        >
          <div
            className="css-15yujhg"
          >
            <svg
              className="chakra-icon css-4i8mdq"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-4i8mdq"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1fdnb51"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1fdnb51"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1fdnb51"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <span
              className="css-ekuaes"
            >
              50
               reviews
            </span>
          </div>
        </td>
        <td
          className="css-0"
          role="gridcell"
        >
          <span
            className="chakra-badge css-ifsvay"
          >
            Active
          </span>
        </td>
        <td
          className="css-0"
          role="gridcell"
        >
          <button
            className="chakra-button css-oqfhsh"
            type="button"
          >
            Restrict
          </button>
        </td>
      </tr>
      <tr
        className="css-0"
        role="row"
      >
        <td
          className="css-0"
          role="gridcell"
        >
          <span
            className="chakra-avatar css-y4jt82"
          >
            <div
              aria-label="user"
              className="chakra-avatar__initials css-0"
            >
              u
            </div>
          </span>
        </td>
        <td
          className="css-0"
          data-testid="name"
          role="gridcell"
        >
          Hirumal
           
          Perera
           
        </td>
        <td
          className="css-0"
          role="gridcell"
        >
          <div
            className="css-15yujhg"
          >
            <svg
              className="chakra-icon css-4i8mdq"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-4i8mdq"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1fdnb51"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1fdnb51"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="chakra-icon css-1fdnb51"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <span
              className="css-ekuaes"
            >
              50
               reviews
            </span>
          </div>
        </td>
        <td
          className="css-0"
          role="gridcell"
        >
          <span
            className="chakra-badge css-ifsvay"
          >
            Active
          </span>
        </td>
        <td
          className="css-0"
          role="gridcell"
        >
          <button
            className="chakra-button css-oqfhsh"
            type="button"
          >
            Restrict
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
